browser/ui/BUILD.gn

```diff
@@ -130,6 +130,8 @@ source_set("ui") {
       "webui/settings/brave_privacy_handler.h",
       "webui/settings/brave_search_engines_handler.cc",
       "webui/settings/brave_search_engines_handler.h",
+      "webui/settings/brave_settings_localized_strings_provider.cc",
```

Comment:
these need to be in a target that is a dep of settings_localized_strings_provider.cc provider is in for this to work correctly

Reply:
yes, `settings_localized_strings_provider.cc` is in the `static_library("ui")` target in `chrome/browser/ui/BUILD.gn` which we patch with `deps+= [ "//brave/browser/ui" ]` (https://github.com/brave/brave-core/blob/master/patches/chrome-browser-ui-BUILD.gn.patch)

Reply:
but this file also has a dep on chrome/browser/ui so it creates yet another circular dep issue in this target

Reply:
let's convert this into source/deps list, because we extend an upstream target, not depend on it.
@bridiver do you have something agains it?

Reply:
yea, I think this most of this needs to move into sources because of the circular deps

Reply:
Ok, then I'll merge this one and fixing deps will be fixed via https://github.com/brave/brave-browser/issues/10614